lib/api/api.js

```diff
@@ -50,6 +51,18 @@ auth.setHandler(vault);
 const api = {
     callApiMethod(apiMethod, request, response, log, callback) {
         let returnTagCount = true;
+
+        const validationRes =
+            validateQueryAndHeaders(request.method, request.query,
+                request.headers, log);
+        if (validationRes.error) {
+            log.debug('request query / header validation failed', {
+                error: validationRes.error,
+                method: 'api.callApiMethod',
+            });
+            return callback(validationRes.error);
```

Comment:
We might need: `proccess.nextTick` here but also for the errors after.
So you might want to add them in another commit.